dotCMS/src/main/java/com/dotcms/jitsu/EventLogWebInterceptor.java

```diff
@@ -0,0 +1,119 @@
+ * <p>
+ * For POST requests to "/api/v1/event", it sends the event in JSON via {@link EventLogSubmitter}
+ */
+
+public class EventLogWebInterceptor implements WebInterceptor {
+    private static final long serialVersionUID = 1L;
+    final String[] paths = new String[] {
+            "/s/lib.js",
+            "/api/v1/event"};
+
+    @Override
+    public String[] getFilters() {
+        return paths;
+    }
+
+    public EventLogWebInterceptor() {
+        submitter = new EventLogSubmitter();
+    }
+
+    final EventLogSubmitter submitter;
+
+    final Lazy<String> jitsuLib = Lazy.of(() -> {
+        try (InputStream in = this.getClass().getResourceAsStream("/jitsu/lib.js")) {
+            return new BufferedReader(
+                    new InputStreamReader(in, StandardCharsets.UTF_8))
+                    .lines()
+                    .collect(Collectors.joining("\n"));
+        } catch (Exception e) {
+            throw new DotRuntimeException(e);
+        }
+    });
+
+    @Override
+    public Result intercept(final HttpServletRequest request, final HttpServletResponse response) throws IOException {
+        setHeaders(response);
+
+        if ("GET".equals(request.getMethod())) {
+            doGet(request, response);
+        }
+
+        if ("POST".equals(request.getMethod())) {
+            Try.run(() -> doPost(request, response));
+        }
+
+        if ("OPTIONS".equals(request.getMethod())) {
+            doOptions(request, response);
+        }
+
+        return Result.SKIP_NO_CHAIN;
+    }
+
+    public void doGet(final HttpServletRequest request, final HttpServletResponse response) throws IOException {
+        Logger.info(getClass(), "doGet");
+        response.getWriter().append(jitsuLib.get());
+    }
+
+    public void doPost(final HttpServletRequest request, final HttpServletResponse response) throws IOException, JSONException {
+        if (request.getContentType() != null && request.getContentType().toLowerCase().indexOf("multipart/form-data") > -1) {
+            return;
+        }
+
+        final String payload = CharStreams.toString(request.getReader());
+        final String realIp = request.getRemoteAddr();
+        JSONObject json = new JSONObject(payload);
+        String mungedIp = (realIp.lastIndexOf(".") > -1) ? realIp.substring(0, realIp.lastIndexOf(".")) + ".1" : "0.0.0.0";
+
+        json.put("ip", mungedIp);
+        json.put("clusterId", ClusterFactory.getClusterId());
+
+        try {
+            this.submitter.logEvent(json.toString());
+        }
+        catch(Exception e) {
+
+
+        }
+    }
+
+    public void doOptions(final HttpServletRequest request, final HttpServletResponse response) {
+        response.setHeader("content-length", "0");
+        response.setStatus(200);
+    }
+
+    public void setHeaders(final HttpServletResponse response) {
+        response.addHeader("content-type", "application/javascript; charset=utf-8");
```

Comment:
Can we use constants from `javax.ws.rs.core.HttpHeaders`

Reply:
Only two of these were available